library/src/main/java/com/github/pwittchen/reactivenetwork/library/ReactiveNetwork.java

```diff
@@ -117,23 +132,42 @@ public static ReactiveNetwork create() {
    */
   public static Observable<Boolean> observeInternetConnectivity(final int intervalInMs,
       final String host, final int port, final int timeoutInMs) {
+    return observeInternetConnectivity(intervalInMs, intervalInMs, host, port, timeoutInMs);
+  }
+
+  /**
+   * Observes connectivity with the Internet by opening socket connection with remote host
+   *
+   * @param initialIntervalInMs in milliseconds determining the delay of the first connectivity check
+   * @param intervalInMs in milliseconds determining how often we want to check connectivity
+   * @param host for checking Internet connectivity
+   * @param port for checking Internet connectivity
+   * @param timeoutInMs for pinging remote host in milliseconds
+   * @return RxJava Observable with Boolean - true, when we have connection with host and false if
+   * not
+   */
+  public static Observable<Boolean> observeInternetConnectivity(final int initialIntervalInMs, final int intervalInMs,
+                                                                final String host, final int port, final int timeoutInMs) {
+    if (initialIntervalInMs < 0) {
+      throw new IllegalArgumentException("initialIntervalInMs is not a non-zero number");
+    }
```

Comment:
Tried doing that but positive checking throws for 0 or lower so had to mirror the preconditions code.